easybuild/easyconfigs/t/Trinity/Trinity-2.9.0-foss-2019a.eb

```diff
@@ -0,0 +1,44 @@
+##
+# This is a contribution from DeepThought HPC Service, Flinders University, Adelaide, Australia
+# Homepage:     https://staff.flinders.edu.au/research/deep-thought
+#
+# Authors::     Robert Qiao <rob.qiao@flinders.edu.au>
+# License::     Custom
+#
+# Notes::
+##
+
+name = 'Trinity'
+version = '2.9.0'
+
+homepage = 'https://trinityrnaseq.github.io'
+description = """Trinity represents a novel method for the efficient and robust de novo reconstruction
+ of transcriptomes from RNA-Seq data. Trinity combines three independent software modules: Inchworm,
+ Chrysalis, and Butterfly, applied sequentially to process large volumes of RNA-Seq reads."""
+
+toolchain = {'name': 'foss', 'version': '2019b'}
+toolchainopts = {'optarch': True}
+
+source_urls = ['https://github.com/trinityrnaseq/trinityrnaseq/releases/download/v%(version)s']
+sources = ['trinityrnaseq-v%(version)s.FULL.tar.gz']
+checksums = ['be09bf8bb4884970349b91e5d48d87b0b2565d361411cd041aeb45b1bd494c28']
+
+builddependencies = [
+    ('Autotools', '20180311', '', ('GCCcore', '8.3.0')),
+    ('CMake', '3.15.3', '', ('GCCcore', '8.3.0')),
+]
+
+dependencies = [
+    ('Java', '11', '', True),
+    ('ant', '1.10.7', '-Java-%(javaver)s', True),
+    ('Bowtie2', '2.3.5.1', '', ('GCC', '8.3.0')),
```

Comment:
@robqiao please remove the subtoolchains GCC, GCCcore and gompi for all dependencies? EB automatically detects this.

Reply:
Thx @smoors for the tips. Will different versions of dependencies also be trialled? Say we have easyconfig file for `Bowtie2-2.3.5.1-GCC-8.2.0` but not `Bowtie-2.3.5.1-GCC-8.2.0`, but we request a version dependency of `GCC-8.3.0` will the build try to compile?

Reply:
@robqiao I don't think so. you could try to first install Bowtie2 with the options `--try-toolchain GCC,8.3.0 --robot`, but there's no guarantee it will work. there is also the option `--try-software-version` to update the version only. for dependencies, all exact variants must be present or installed.
there is work being to improve this, see https://github.com/easybuilders/easybuild-framework/pull/2599